torch/_inductor/triton_heuristics.py

```diff
@@ -165,10 +166,56 @@ def precompile(self, warm_cache_only_with_cc=None):
         with self.lock:
             if self.launchers:
                 return
-            self.launchers = [
-                self._precompile_config(c, warm_cache_only_with_cc)
-                for c in self.configs
-            ]
+            self.launchers = []
+            compiled_binaries = []
+            for c in self.configs:
+                compiled_binary, launcher = self._precompile_config(
+                    c, warm_cache_only_with_cc
+                )
+                self.launchers.append(launcher)
+                compiled_binaries.append(compiled_binary)
+
+            if (
+                config.dynamic_scale_rblock
+                and self.heuristic_type == HeuristicType.REDUCTION
```

Comment:
PERSISTENT_REDUCTION has an assumption that RBLOCK = next_power_of_two(rnumel).

If we want to reduce RBLOCK, we should convert PERSISTENT_REDUCTION to REDUCTION in the first place. The 'multi-kernel' work should help here since we can use autotune to pick between REDUCTION and PERSISTENT_REDUCTION.